hack/test-cmd.sh

```diff
@@ -699,7 +699,7 @@ __EOF__
   # Command: autoscale rc "frontend"
   kubectl autoscale -f examples/guestbook/frontend-controller.yaml --save-config "${kube_flags[@]}" --max=2
   # Post-Condition: hpa "frontend" has configuration annotation
-  [[ "$(kubectl get hpa frontend -o yaml "${kube_flags[@]}" | grep kubectl.kubernetes.io/last-applied-configuration)" ]]
```

Comment:
> kind.group seems unnatural to me but I guess it's too late to change?

We talked about it for a while here: https://github.com/kubernetes/kubernetes/pull/14461#issuecomment-147833111.  Groups as qualified domain names give natural and identifiable namespacing.  In that light, having "resource.group.domain.name" is a natural qualification.

This pull establishes all the necessary parts to even support your suggested "resource.version.group".  Since that would be fully specified, it can be built on these primitives as completed by discovery (still on my wishlist, but not for 1.2).